en/03_Drawing_a_triangle/00_Setup/01_Instance.adoc

```diff
@@ -16,51 +19,55 @@ void initVulkan() {
 }
 ----
 
-Additionally add a data member to hold the handle to the instance:
+Additionally, add a data member to hold the handle to the instance and the
+raii context:
 
 [,c++]
 ----
 private:
-VkInstance instance;
+vk::raii::Context  context;
+std::unique_ptr<vk::raii::Instance> instance;
 ----
 
-Now, to create an instance we'll first have to fill in a struct with some information about our application.
-This data is technically optional, but it may provide some useful information to the driver in order to optimize our specific application (e.g.
-because it uses a well-known graphics engine with certain special behavior).
-This struct is called `VkApplicationInfo`:
+Now, to create an instance, we'll first have to fill in a struct with some
+information about our application. This data is technically optional, but it may
+provide some useful information to the driver to optimize our specific
+ application, (e.g., because it uses a well-known graphics engine with
+certain special behavior). This struct is called `VkApplicationInfo`:
 
 [,c++]
 ----
 void createInstance() {
-    VkApplicationInfo appInfo{};
-    appInfo.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
-    appInfo.pApplicationName = "Hello Triangle";
-    appInfo.applicationVersion = VK_MAKE_VERSION(1, 0, 0);
-    appInfo.pEngineName = "No Engine";
-    appInfo.engineVersion = VK_MAKE_VERSION(1, 0, 0);
-    appInfo.apiVersion = VK_API_VERSION_1_0;
+    constexpr auto appInfo = vk::ApplicationInfo("Hello Triangle", 1, "No  Engine", 1, vk::ApiVersion14);
 }
 ----
 
-As mentioned before, many structs in Vulkan require you to explicitly specify the type in the `sType` member.
-This is also one of the many structs with a `pNext` member that can point to extension information in the future.
-We're using value initialization here to leave it as `nullptr`.
+While vk::ApiVersion10 or Vulkan 1.0 does exist, some functionality
+is older and doesn't work well with RAII, or as we'll link:../02_Graphics_pipeline_basics/01_Shader_modules.adoc[talk about later],
+ the SLang language, so we're showing 1.4 as our baseline. If we were in the C
```

Comment:
Should be Slang instead of SLang, at least accordig to [Slang](https://github.com/shader-slang/slang)